Add collections feature (#1214)
memgraph/documentation: pages/memgraph-lab/features/collections.mdx

---
title: Collections
description: A feature that lets you save and organize your favorite queries 
---

import {CommunityLinks} from '/components/social-card/CommunityLinks'
import { Steps } from 'nextra/components'

# Collections

​In Memgraph Lab, the collections feature allows you to organize and manage your
Cypher queries efficiently by grouping them into named collections. This
functionality is particularly useful for structuring queries related to specific
projects, datasets, or analytical tasks, enhancing both organization and
collaboration.​

> ## Key features
> 
> - **Organized query management**: Group related queries together to streamline your
> workflow and quickly locate specific queries when needed.​
> 
> - **Enhanced collaboration**: Share collections with team members, facilitating better
> collaboration and ensuring consistency across projects.​ 
> 
> - **Efficient query execution**: Execute queries directly from their respective
> collections, reducing redundancy and improving productivity.​

Reviewer: Just a suggestion, but I would incorporate this section into the previous one and not have Key features as a heading
```suggestion
With the collections feature, you can: 
- **Organize query management**: Group related queries together to streamline your
workflow and quickly locate specific queries when needed.​

- **Enhance collaboration**: Share collections with team members, facilitating better
collaboration and ensuring consistency across projects.​ 

- **Efficiently execute queries**: Execute queries directly from their respective
collections, reducing redundancy and improving productivity.​
```

![add-to-collection](/pages/data-visualization/features/collections/add-to-collection.png)

You can either choose an **existing collection** from the dropdown or **create a new
one** on the spot by entering a name during the flow.
You'll also need to provide a **query name** before saving.

![query-name-collections](/pages/data-visualization/features/collections/query-name-collections.png)

### Add query from the Run history

You can revisit previously executed queries and add them to a collection.

![add-from-run-history](/pages/data-visualization/features/collections/add-from-run-history.png)

You can either choose an **existing collection** from the dropdown or **create a
new one** on the spot by entering a name during the flow.

### Add query from the Collections

If you're managing collections directly, simply select a collection and click
**Add query** to manually write and add a new query to the currently open
collection.

## Importing and exporting collections

Query collections can be exported and imported as JSON files, making it easy to
share across teams or back up your work.

![collection-export](/pages/data-visualization/features/collections/collection-export.png)

> ## Pre-prepared collections 

Reviewer: This section was a bit unclear to me, by reading I would assume I have pre-prepared collection as soon as I start Lab. I would add sooner that it correlates to the chosen dataset
